packages/sdk-core/src/modules/metrics/MetricsSubmissionQueue.ts

```diff
@@ -0,0 +1,82 @@
+import { Delay } from '../../common/DelayHelper';
+import { TimeHelper } from '../../common/TimeHelper';
+import { BacktraceRequestHandler } from '../../model/http';
+import { MetricsQueue } from './MetricsQueue';
+import { MetricsEvent } from './model/MetricsEvent';
+
+export class MetricsSubmissionQueue<T extends MetricsEvent> implements MetricsQueue<T> {
+    public get total() {
+        return this._events.length;
+    }
+
+    public get submissionUrl() {
+        return this._submissionUrl;
+    }
+
+    public readonly TIMEOUT_BETWEEN_REQUESTS = TimeHelper.convertSecondsToMilliseconds(10);
+
+    private _events: T[] = [];
+    private _numberOfDroppedRequests = 0;
+
+    private readonly MAXIMUM_NUMBER_OF_ATTEMPTS = 3;
+
+    constructor(
+        private readonly _submissionUrl: string,
+        private readonly _eventName: string,
+        private readonly _requestHandler: BacktraceRequestHandler,
+        private readonly _metricMetadata: Record<string, unknown>,
+        public readonly maximumEvents: number = 50,
+    ) {}
+
+    public add(event: T) {
+        this._events.push(event);
+        if (this.reachedLimit()) {
+            this.send();
+        }
+    }
+
+    public async send() {
+        const eventsToProcess = this._events.splice(0);
+        return await this.submitMetricsToServer(eventsToProcess);
+    }
```

Comment:
Is there a chance that the `send` method is executed before the previous send completed? Maybe we should limit this somehow?

Reply:
Yes, it could happen. In the worst-case scenario if there are more events in the queue than maximumEvents we can lose some events. Unfortunately, there is no perfect way to do this. We can also make the interface asynchronous, but one of the requirements we got for metrics is to send them when we reach the limit. There is a tradeoff I agree. 

let me back to you with some ideas how we can change it